File: android/demos/tdd-basics-demo/src/main/kotlin/SpeakingClock.kt
class SpeakingClock(
        private val clock: Clock,
        private val speechEngine: SpeechEngine,
        private val converter: TimeToTextConverter
) {
    fun sayTime() {
        val time = clock.getTime()
        val timeAsText = converter.convertTimeToText(time)
        speechEngine.speak(timeAsText)
    }
}

File: ios/demos/tdd-basics-demo/Sources/SpeakingClock/SpeakingClock.swift
import Foundation

class SpeakingClock {
    private let clock: Clock
    private let speechSynthesizer: SpeechSynthesizer
    private let converter: TimeToTextConverter

    init(clock: Clock, speechSynthesizer: SpeechSynthesizer, converter: TimeToTextConverter) {
        self.clock = clock
        self.speechSynthesizer = speechSynthesizer
        self.converter = converter
    }

    func sayTime() {
        let time = clock.getTime()
        if let timeAsText = converter.convertTimeToText(time) {
            speechSynthesizer.speak(timeAsText)
        }
    }
}